Fix linter bugs for new pylint version (`1.7.0`)

The new `pylint` version (`1.7.0`), recently released (~3 hours ago), is catching more errors on our code base.
We need to review our code in order to be in compliance with the tests.

The current output of the tests are:
```
py36 runtests: commands[3] | pylama tests setup.py pyof
tests/test_foundation/test_basic_types.py:21:0: W0511 TODO [pylint]
tests/test_foundation/test_basic_types.py:27:0: W0511 TODO [pylint]
tests/test_foundation/test_basic_types.py:45:0: W0511 TODO [pylint]
tests/test_foundation/test_basic_types.py:51:0: W0511 TODO [pylint]
tests/test_foundation/test_basic_types.py:69:0: W0511 TODO [pylint]
tests/test_foundation/test_basic_types.py:75:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_flow_match.py:40:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_flow_match.py:46:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_queue.py:23:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_queue.py:29:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_queue.py:49:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_queue.py:55:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_queue.py:74:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_queue.py:80:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_phy_port.py:32:0: W0511 TODO [pylint]
tests/v0x01/test_common/test_phy_port.py:38:0: W0511 TODO [pylint]
pyof/v0x01/common/flow_match.py:174:15: R0123 Comparison to literal [pylint]
pyof/v0x01/controller2switch/barrier_request.py:23:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x01/controller2switch/get_config_request.py:16:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x01/controller2switch/stats_reply.py:22:5: E303 too many blank lines (2) [pycodestyle]
pyof/v0x01/controller2switch/stats_reply.py:83:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/v0x01/controller2switch/features_request.py:20:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x01/controller2switch/barrier_reply.py:23:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x01/controller2switch/features_reply.py:82:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x01/symmetric/hello.py:23:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x04/common/table_feature.py:66:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/v0x04/controller2switch/barrier_request.py:23:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x04/controller2switch/get_config_request.py:16:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x04/controller2switch/features_request.py:20:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x04/controller2switch/barrier_reply.py:23:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x04/controller2switch/common.py:96:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/v0x04/controller2switch/features_reply.py:86:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/v0x04/controller2switch/get_async_request.py:23:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/foundation/base.py:253:4: R0914 Too many local variables (18/15) [pylint]
pyof/foundation/base.py:253:4: C0204 Metaclass class method __new__ should have 'mcs' as first argument [pylint]
pyof/foundation/base.py:291:38: W0212 Access to a protected member _get_class_attributes of a client class [pylint]
pyof/foundation/base.py:297:0: C0330 Wrong continued indentation (remove 1 space).                                                                )                                                               |^ [pylint]
pyof/foundation/base.py:375:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/foundation/base.py:488:15: R1701 Consider merging these isinstance calls to isinstance(obj, (GenericStruct, GenericType)) [pylint]
pyof/foundation/base.py:564:11: C1801 Do not use `len(SEQUENCE)` as condition value [pylint]
pyof/foundation/network_types.py:135:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/foundation/basic_types.py:319:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/foundation/basic_types.py:410:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/foundation/basic_types.py:433:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/foundation/basic_types.py:433:40: C1801 Do not use `len(SEQUENCE)` as condition value [pylint]
pyof/foundation/basic_types.py:461:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/foundation/basic_types.py:509:24: E1120 No value for argument 'items' in constructor call [pylint]
pyof/foundation/basic_types.py:554:8: R1705 Unnecessary "else" after "return" [pylint]
pyof/foundation/basic_types.py:555:12: R1705 Unnecessary "else" after "return" [pylint]
pyof/foundation/basic_types.py:555:15: C1801 Do not use `len(SEQUENCE)` as condition value [pylint]
pyof/foundation/basic_types.py:647:4: W0235 Useless super delegation in method '__init__' [pylint]
pyof/foundation/basic_types.py:671:13: C1801 Do not use `len(SEQUENCE)` as condition value [pylint]
pyof/foundation/basic_types.py:690:11: C1801 Do not use `len(SEQUENCE)` as condition value [pylint]
ERROR: InvocationError: '/home/diraol/devel/python-openflow/.tox/py36/bin/pylama tests setup.py pyof'

```

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -250,7 +250,7 @@ class MetaStruct(type):
     def __prepare__(mcs, name, bases, **kwargs):
         return OrderedDict()
 
-    def __new__(cls, name, bases, classdict, **kwargs):
+    def __new__(mcs, name, bases, classdict, **kwargs):
         """Inherit attributes from parent class and update their versions.
 
         Here is the moment that the new class is going to be created. During
@@ -330,7 +330,7 @@ class MetaStruct(type):
             inherited_attributes.update(classdict)
             classdict = inherited_attributes
 
-        return super().__new__(cls, name, bases, classdict, **kwargs)
+        return super().__new__(mcs, name, bases, classdict, **kwargs)
 
     @staticmethod
     def get_pyof_version(module_fullname):

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -374,8 +374,8 @@ class MetaStruct(type):
         module_version = MetaStruct.get_pyof_version(module_fullname)
         if not module_version or module_version == version:
             return None
-        else:
-            return module_fullname.replace(module_version, version)
+
+        return module_fullname.replace(module_version, version)
 
     @staticmethod
     def get_pyof_obj_new_version(name, obj, new_version):

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -561,7 +561,7 @@ class GenericStruct(object, metaclass=MetaStruct):
     def _unpack_attribute(self, name, obj, buff, begin):
         attribute = deepcopy(obj)
         setattr(self, name, attribute)
-        if len(buff) == 0:
+        if not buff:
             size = 0
         else:
             try:

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -296,20 +296,7 @@ class MetaStruct(type):
                                                                curr_version)
 
                     if attr_name == 'header':
-                        #: Here we are going to set the message_type on the
-                        #: header, according to the message_type of the
-                        #: parent class.
-                        old_enum = obj.message_type
-                        new_header = attr[1]
-                        new_enum = new_header.__class__.message_type.enum_ref
-                        #: This 'if' will be removed on the future with an
-                        #: improved version of __init_subclass__ method of the
-                        #: GenericMessage class
-                        if old_enum:
-                            msg_type_name = old_enum.name
-                            new_type = new_enum[msg_type_name]
-                            new_header.message_type = new_type
-                        attr = (attr[0], new_header)
+                        attr = mcs._header_message_type_update(obj, attr)
 
                     inherited_attributes.update([attr])
                 #: We are going to inherit just from the 'closest parent'
@@ -330,6 +317,25 @@ class MetaStruct(type):
             classdict = inherited_attributes
 
         return super().__new__(mcs, name, bases, classdict, **kwargs)
+
+    @staticmethod
+    def _header_message_type_update(obj, attr):
+        """Update the message type on the header.
+
+        Set the message_type of the header according to the message_type of
+        the parent class.
+        """
+        old_enum = obj.message_type
+        new_header = attr[1]
+        new_enum = new_header.__class__.message_type.enum_ref
+        #: This 'if' will be removed on the future with an
+        #: improved version of __init_subclass__ method of the
+        #: GenericMessage class
+        if old_enum:
+            msg_type_name = old_enum.name
+            new_type = new_enum[msg_type_name]
+            new_header.message_type = new_type
+        return (attr[0], new_header)
 
     @staticmethod
     def get_pyof_version(module_fullname):

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -293,8 +293,7 @@ class MetaStruct(type):
                     #: considering the version of the current class being
                     #: created.
                     attr = MetaStruct.get_pyof_obj_new_version(attr_name, obj,
-                                                               curr_version
-                                                               )
+                                                               curr_version)
 
                     if attr_name == 'header':
                         #: Here we are going to set the message_type on the

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -288,7 +288,7 @@ class MetaStruct(type):
             #: Check if we are inheriting from one of our classes.
             if isinstance(base, MetaStruct):
                 inherited_attributes = OrderedDict()
-                for attr_name, obj in base._get_class_attributes():
+                for attr_name, obj in base.get_class_attributes():
                     #: Get an updated version of this attribute,
                     #: considering the version of the current class being
                     #: created.
@@ -456,7 +456,7 @@ class GenericStruct(object, metaclass=MetaStruct):
 
     def __init__(self):
         """Contructor takes no argument and stores attributes' deep copies."""
-        for name, value in self._get_class_attributes():
+        for name, value in self.get_class_attributes():
             setattr(self, name, deepcopy(value))
 
     def __eq__(self, other):
@@ -505,7 +505,7 @@ class GenericStruct(object, metaclass=MetaStruct):
         return True
 
     @classmethod
-    def _get_class_attributes(cls):
+    def get_class_attributes(cls):
         """Return a generator for class attributes' names and value.
 
         This method strict relies on the PEP 520 (Preserving Class Attribute
@@ -516,9 +516,9 @@ class GenericStruct(object, metaclass=MetaStruct):
 
         .. code-block:: python3
 
-            for _name, _value in self._get_class_attributes():
-                print("attribute name: {}".format(_name))
-                print("attribute type: {}".format(_value))
+            for name, value in self.get_class_attributes():
+                print("attribute name: {}".format(name))
+                print("attribute type: {}".format(value))
 
         returns:
             generator: tuples with attribute name and value.
@@ -544,7 +544,7 @@ class GenericStruct(object, metaclass=MetaStruct):
             generator: tuples with attribute name and value.
         """
         for name, value in self.__dict__.items():
-            if name in map((lambda x: x[0]), self._get_class_attributes()):
+            if name in map((lambda x: x[0]), self.get_class_attributes()):
                 yield (name, value)
 
     def _get_attributes(self):
@@ -561,7 +561,7 @@ class GenericStruct(object, metaclass=MetaStruct):
         """
         return map((lambda i, c: (i[1], c[1])),
                    self._get_instance_attributes(),
-                   self._get_class_attributes())
+                   self.get_class_attributes())
 
     def _unpack_attribute(self, name, obj, buff, begin):
         attribute = deepcopy(obj)
@@ -646,7 +646,7 @@ class GenericStruct(object, metaclass=MetaStruct):
             offset (int): Where to begin unpacking.
         """
         begin = offset
-        for name, value in self._get_class_attributes():
+        for name, value in self.get_class_attributes():
             size = self._unpack_attribute(name, value, buff, begin)
             begin += size
 
@@ -750,7 +750,7 @@ class GenericMessage(GenericStruct):
             offset (int): Where to begin unpacking.
         """
         begin = offset
-        for name, value in self._get_class_attributes():
+        for name, value in self.get_class_attributes():
             if type(value).__name__ != "Header":
                 size = self._unpack_attribute(name, value, buff, begin)
                 begin += size

--- a/pyof/foundation/base.py
+++ b/pyof/foundation/base.py
@@ -485,7 +485,7 @@ class GenericStruct(object, metaclass=MetaStruct):
             True: if the obj is a kytos attribute
             False: if the obj is not a kytos attribute
         """
-        return isinstance(obj, GenericType) or isinstance(obj, GenericStruct)
+        return isinstance(obj, (GenericType, GenericStruct))
 
     def _validate_attributes_type(self):
         """Validate the type of each attribute."""

--- a/pyof/foundation/basic_types.py
+++ b/pyof/foundation/basic_types.py
@@ -560,7 +560,7 @@ class TypeList(list, GenericStruct):
                 # length of the list multiplied by the size of the GenericType.
                 return len(self) * self[0].get_size()
 
-            #Otherwise iter over the list accumulating the sizes.
+            # Otherwise iter over the list accumulating the sizes.
             return sum(item.get_size() for item in self)
 
         return type(self)(value).get_size()

--- a/pyof/foundation/basic_types.py
+++ b/pyof/foundation/basic_types.py
@@ -316,7 +316,7 @@ class IPAddress(GenericType):
 class HWAddress(GenericType):
     """Defines a hardware address."""
 
-    def __init__(self, hw_address='00:00:00:00:00:00'):
+    def __init__(self, hw_address='00:00:00:00:00:00'):  # noqa
         """The constructor takes the parameters below.
 
         Args:
@@ -407,7 +407,7 @@ class BinaryData(GenericType):
     return the size of the instance using Python's :func:`len`.
     """
 
-    def __init__(self, value=b''):
+    def __init__(self, value=b''):  # noqa
         """The constructor takes the parameter below.
 
         Args:
@@ -644,7 +644,7 @@ class ConstantTypeList(TypeList):
     list operations.
     """
 
-    def __init__(self, items=None):
+    def __init__(self, items=None):  # noqa
         """The contructor can contain the items to be stored.
 
         Args:

--- a/pyof/foundation/basic_types.py
+++ b/pyof/foundation/basic_types.py
@@ -432,8 +432,8 @@ class BinaryData(GenericType):
 
         if isinstance(value, bytes) and len(value) > 0:
             return value
-        else:
-            return b''
+
+        return b''
 
     def unpack(self, buff, offset=0):
         """Unpack a binary message into this object's attributes.
@@ -462,8 +462,8 @@ class BinaryData(GenericType):
             return len(self._value)
         elif hasattr(value, 'get_size'):
             return value.get_size()
-        else:
-            return len(value)
+
+        return len(value)
 
 
 class TypeList(list, GenericStruct):
@@ -559,11 +559,11 @@ class TypeList(list, GenericStruct):
                 # If the type of the elements is GenericType, then returns the
                 # length of the list multiplied by the size of the GenericType.
                 return len(self) * self[0].get_size()
-            else:
-                # Otherwise iter over the list accumulating the sizes.
-                return sum(item.get_size() for item in self)
-        else:
-            return type(self)(value).get_size()
+
+            #Otherwise iter over the list accumulating the sizes.
+            return sum(item.get_size() for item in self)
+
+        return type(self)(value).get_size()
 
     def __str__(self):
         """Human-readable object representantion."""

--- a/pyof/foundation/basic_types.py
+++ b/pyof/foundation/basic_types.py
@@ -430,7 +430,7 @@ class BinaryData(GenericType):
         if value is None:
             value = self._value
 
-        if isinstance(value, bytes) and len(value) > 0:
+        if isinstance(value, bytes) and value:
             return value
 
         return b''
@@ -552,7 +552,7 @@ class TypeList(list, GenericStruct):
             int: The size in bytes.
         """
         if value is None:
-            if len(self) == 0:
+            if not self:
                 # If this is a empty list, then returns zero
                 return 0
             elif issubclass(type(self[0]), GenericType):
@@ -668,7 +668,7 @@ class ConstantTypeList(TypeList):
         """
         if isinstance(item, list):
             self.extend(item)
-        elif len(self) == 0:
+        elif not self:
             list.append(self, item)
         elif item.__class__ == self[0].__class__:
             list.append(self, item)
@@ -687,7 +687,7 @@ class ConstantTypeList(TypeList):
             :exc:`~.exceptions.WrongListItemType`: If an item has a different
                 type than the first item to be stored.
         """
-        if len(self) == 0:
+        if not self:
             list.append(self, item)
         elif item.__class__ == self[0].__class__:
             list.insert(self, index, item)

--- a/pyof/foundation/basic_types.py
+++ b/pyof/foundation/basic_types.py
@@ -506,7 +506,7 @@ class TypeList(list, GenericStruct):
         if value is None:
             value = self
         else:
-            container = type(self)()
+            container = type(self)(items=None)
             container.extend(value)
             value = container
 

--- a/pyof/foundation/network_types.py
+++ b/pyof/foundation/network_types.py
@@ -134,8 +134,8 @@ class GenericTLV(GenericStruct):
         """Return struct size."""
         if isinstance(value, type(self)):
             return value.get_size()
-        else:
-            return 2 + self.length
+
+        return 2 + self.length
 
 
 class TLVWithSubType(GenericTLV):

--- a/pyof/v0x01/common/flow_match.py
+++ b/pyof/v0x01/common/flow_match.py
@@ -171,7 +171,7 @@ class Match(GenericStruct):
 
         default_value = getattr(Match, field)
         if isinstance(default_value, IPAddress):
-            if field is 'nw_dst':
+            if field == 'nw_dst':
                 self.wildcards |= FlowWildCards.OFPFW_NW_DST_MASK
                 shift = FlowWildCards.OFPFW_NW_DST_SHIFT
             else:

--- a/pyof/v0x01/controller2switch/barrier_reply.py
+++ b/pyof/v0x01/controller2switch/barrier_reply.py
@@ -19,11 +19,3 @@ class BarrierReply(GenericMessage):
     """
 
     header = Header(message_type=Type.OFPT_BARRIER_REPLY)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        Args:
-            xid (int): Header's xid.
-        """
-        super().__init__(xid)

--- a/pyof/v0x01/controller2switch/barrier_request.py
+++ b/pyof/v0x01/controller2switch/barrier_request.py
@@ -19,11 +19,3 @@ class BarrierRequest(GenericMessage):
     """
 
     header = Header(message_type=Type.OFPT_BARRIER_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x01/controller2switch/features_reply.py
+++ b/pyof/v0x01/controller2switch/features_reply.py
@@ -78,21 +78,3 @@ class SwitchFeatures(GenericMessage):
 
 class FeaturesReply(SwitchFeatures):
     """'OFPT_FEATURES_REPLY' message."""
-
-    def __init__(self, xid=None, datapath_id=None, n_buffers=None,
-                 n_tables=None, capabilities=None, actions=None, ports=None):
-        """The constructor just assings parameters to object attributes.
-
-        Args:
-            xid (int): xid to be used on the message header.
-            datapath_id (str or :class:`.DPID`): DPID datapath unique ID.
-                The lower 48-bits are for MAC address, while
-                the upper 16-bits are implementer-defined.
-            n_buffers (int): UBInt32 max packets buffered at once.
-            n_tables (int): UBInt8 number of tables supported by datapath.
-            capabilities (int): UBInt32 bitmap of supported capabilities.
-            actions (int): UBInt32 Bitmap of supported "action_type"s.
-            ports (int): Port definitions.
-        """
-        super().__init__(xid, datapath_id, n_buffers, n_tables, capabilities,
-                         actions, ports)

--- a/pyof/v0x01/controller2switch/features_request.py
+++ b/pyof/v0x01/controller2switch/features_request.py
@@ -16,11 +16,3 @@ class FeaturesRequest(GenericMessage):
 
     header = Header(
         message_type=Type.OFPT_FEATURES_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor takes the optional parameter below.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x01/controller2switch/get_config_request.py
+++ b/pyof/v0x01/controller2switch/get_config_request.py
@@ -12,13 +12,3 @@ class GetConfigRequest(GenericMessage):
     """Get Config Request message."""
 
     header = Header(message_type=Type.OFPT_GET_CONFIG_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        This message does not contain a body beyond the OpenFlow Header.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x01/controller2switch/packet_out.py
+++ b/pyof/v0x01/controller2switch/packet_out.py
@@ -91,7 +91,7 @@ class PacketOut(GenericMessage):
             offset (int): Where to begin unpacking.
         """
         begin = offset
-        for attribute_name, class_attribute in self._get_class_attributes():
+        for attribute_name, class_attribute in self.get_class_attributes():
             if type(class_attribute).__name__ != "Header":
                 attribute = deepcopy(class_attribute)
                 if attribute_name == 'actions':

--- a/pyof/v0x01/controller2switch/stats_reply.py
+++ b/pyof/v0x01/controller2switch/stats_reply.py
@@ -18,7 +18,6 @@ class StatsReply(GenericMessage):
     flags = UBInt16()
     body = BinaryData()
 
-
     def __init__(self, xid=None, body_type=None, flags=None, body=b''):
         """The constructor just assings parameters to object attributes.
 

--- a/pyof/v0x01/controller2switch/stats_reply.py
+++ b/pyof/v0x01/controller2switch/stats_reply.py
@@ -84,8 +84,8 @@ class StatsReply(GenericMessage):
             return BinaryData(b'')
         elif pyof_class is DescStats:
             return pyof_class()
-        else:
-            return FixedTypeList(pyof_class=pyof_class)
+
+        return FixedTypeList(pyof_class=pyof_class)
 
     def _get_body_class(self):
         if isinstance(self.body_type, (int, UBInt16)):

--- a/pyof/v0x01/symmetric/hello.py
+++ b/pyof/v0x01/symmetric/hello.py
@@ -19,11 +19,3 @@ class Hello(GenericMessage):
     """
 
     header = Header(message_type=Type.OFPT_HELLO, length=8)
-
-    def __init__(self, xid=None):
-        """The constructor takes the parameters below.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x04/common/table_feature.py
+++ b/pyof/v0x04/common/table_feature.py
@@ -69,8 +69,8 @@ class TableFeaturePropType(Enum):
             return NextTablesProperty
         elif self.value <= 7:
             return ActionsProperty
-        else:
-            return OxmProperty
+
+        return OxmProperty
 
 
 # Classes

--- a/pyof/v0x04/controller2switch/barrier_reply.py
+++ b/pyof/v0x04/controller2switch/barrier_reply.py
@@ -19,11 +19,3 @@ class BarrierReply(GenericMessage):
     """
 
     header = Header(message_type=Type.OFPT_BARRIER_REPLY)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        Args:
-            xid (int): Header's xid.
-        """
-        super().__init__(xid)

--- a/pyof/v0x04/controller2switch/barrier_request.py
+++ b/pyof/v0x04/controller2switch/barrier_request.py
@@ -19,11 +19,3 @@ class BarrierRequest(GenericMessage):
     """
 
     header = Header(message_type=Type.OFPT_BARRIER_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x04/controller2switch/common.py
+++ b/pyof/v0x04/controller2switch/common.py
@@ -99,8 +99,8 @@ class TableFeaturePropType(Enum):
             return NextTablesProperty
         elif self.value <= 7:
             return ActionsProperty
-        else:
-            return OxmProperty
+
+        return OxmProperty
 
 
 class MultipartTypes(Enum):

--- a/pyof/v0x04/controller2switch/features_reply.py
+++ b/pyof/v0x04/controller2switch/features_reply.py
@@ -82,22 +82,3 @@ class SwitchFeatures(GenericMessage):
 
 class FeaturesReply(SwitchFeatures):
     """'OFPT_FEATURES_REPLY' message."""
-
-    def __init__(self, xid=None, datapath_id=None, n_buffers=None,
-                 n_tables=None, auxiliary_id=None, capabilities=None,
-                 reserved=None):
-        """The constructor just assings parameters to object attributes.
-
-        Args:
-            xid (int): xid to be used on the message header.
-            datapath_id (int): Datapath unique ID.
-                The lower 48-bits are for MAC address, while
-                the upper 16-bits are implementer-defined.
-            n_buffers (int): Max packets buffered at once.
-            n_tables (int): Number of tables supported by datapath.
-            auxiliary_id (int): Identify auxiliary connections.
-            capabilities (int): bitmap of supported capabilities.
-            reserved (int): Reserved.
-        """
-        super().__init__(xid, datapath_id, n_buffers, n_tables, auxiliary_id,
-                         capabilities, reserved)

--- a/pyof/v0x04/controller2switch/features_request.py
+++ b/pyof/v0x04/controller2switch/features_request.py
@@ -16,11 +16,3 @@ class FeaturesRequest(GenericMessage):
 
     header = Header(
         message_type=Type.OFPT_FEATURES_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor takes the optional parameter below.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x04/controller2switch/get_async_request.py
+++ b/pyof/v0x04/controller2switch/get_async_request.py
@@ -19,13 +19,3 @@ class GetAsyncRequest(GenericMessage):
 
     #: OpenFlow :class:`~.header.Header`
     header = Header(message_type=Type.OFPT_GET_ASYNC_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        This message does not contain a body beyond the OpenFlow Header.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/pyof/v0x04/controller2switch/get_config_request.py
+++ b/pyof/v0x04/controller2switch/get_config_request.py
@@ -12,13 +12,3 @@ class GetConfigRequest(GenericMessage):
     """Get Config Request message."""
 
     header = Header(message_type=Type.OFPT_GET_CONFIG_REQUEST)
-
-    def __init__(self, xid=None):
-        """The constructor just assings parameters to object attributes.
-
-        This message does not contain a body beyond the OpenFlow Header.
-
-        Args:
-            xid (int): xid to be used on the message header.
-        """
-        super().__init__(xid)

--- a/tests/test_class_inheritance/test_inheritance.py
+++ b/tests/test_class_inheritance/test_inheritance.py
@@ -41,11 +41,9 @@ class TestInheritance(unittest.TestCase):
         m1 = self.MyClassA()
         m2 = self.MyClassB()
         # Checking keys (attributes names) and its ordering
-        # pylint: disable=W0212
-        self.assertEqual([attr[0] for attr in m1._get_class_attributes()],
+        self.assertEqual([attr[0] for attr in m1.get_class_attributes()],
                          self.a_expected_names)
-        # pylint: disable=W0212
-        self.assertEqual([attr[0] for attr in m2._get_class_attributes()],
+        self.assertEqual([attr[0] for attr in m2.get_class_attributes()],
                          self.b_expected_names)
 
         # Check if there is no shared attribute between instances

--- a/tests/test_foundation/test_basic_types.py
+++ b/tests/test_foundation/test_basic_types.py
@@ -18,13 +18,11 @@ class TestUBInt8(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Foundation/BasicTypes/UBInt8] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Foundation/BasicTypes/UBInt8] - unpacking."""
-        # TODO
         pass
 
 
@@ -42,13 +40,11 @@ class TestUBInt16(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Foundation/BasicTypes/UBInt16] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Foundation/BasicTypes/UBInt16] - unpacking."""
-        # TODO
         pass
 
 
@@ -66,13 +62,11 @@ class TestUBInt32(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Foundation/BasicTypes/UBInt32] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Foundation/BasicTypes/UBInt32] - unpacking."""
-        # TODO
         pass
 
 

--- a/tests/v0x01/test_common/test_flow_match.py
+++ b/tests/v0x01/test_common/test_flow_match.py
@@ -37,11 +37,9 @@ class TestMatch(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Common/FlowMatch] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Common/FlowMatch] - unpacking."""
-        # TODO
         pass

--- a/tests/v0x01/test_common/test_phy_port.py
+++ b/tests/v0x01/test_common/test_phy_port.py
@@ -29,11 +29,9 @@ class TestPhyPort(TestCase):
     @skip('Not yet implemented')
     def test_pack(self):
         """[Common/PhyPort] - packing."""
-        # TODO
         pass
 
     @skip('Not yet implemented')
     def test_unpack(self):
         """[Common/PhyPort] - unpacking."""
-        # TODO
         pass

--- a/tests/v0x01/test_common/test_queue.py
+++ b/tests/v0x01/test_common/test_queue.py
@@ -20,13 +20,11 @@ class TestQueuePropHeader(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Common/QueuePropHeader] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Common/QueuePropHeader] - unpacking."""
-        # TODO
         pass
 
 
@@ -46,13 +44,11 @@ class TestPacketQueue(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Common/PacketQueue] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Common/PacketQueue] - unpacking."""
-        # TODO
         pass
 
 
@@ -71,11 +67,9 @@ class TestQueuePropMinRate(unittest.TestCase):
     @unittest.skip('Not yet implemented')
     def test_pack(self):
         """[Common/PropMinRate] - packing."""
-        # TODO
         pass
 
     @unittest.skip('Not yet implemented')
     def test_unpack(self):
         """[Common/PropMinRate] - unpacking."""
-        # TODO
         pass

